packages/broker/test/integration/multiple-publisher-plugins.test.ts

```diff
@@ -0,0 +1,189 @@
+        return this.client!.send(JSON.stringify(msg))
+    }
+    async close(): Promise<void> {
+        return this.client!.close()
+    }
+}
+
+class HttpPluginPublisher implements PluginPublisher {
+    async connect(): Promise<void> {
+    }
+    async publish(msg: object, streamId: string): Promise<unknown> {
+        return sendPostRequest(`http://localhost:${httpPort}/streams/${encodeURIComponent(streamId)}`, msg)
+    }
+    async close(): Promise<void> {
+    }
+}
+
+const publishMessages = async (streamId: string): Promise<any[]> => {
+    const publishers: Record<string,PluginPublisher> = {
+        mqtt1: new MqttPluginPublisher(),
+        mqtt2: new MqttPluginPublisher(),
+        websocket1: new WebsocketPluginPublisher(),
+        websocket2: new WebsocketPluginPublisher(),
+        http1: new HttpPluginPublisher(),
+        http2: new HttpPluginPublisher(),
+    }
+    await Promise.all(Object.values(publishers).map((publisher) => publisher.connect(streamId)))
+    const messages: { index: number, publisher: string }[] = []
+    for (const index of range(MESSAGE_COUNT)) {
+        messages.push({
+            index,
+            publisher: sample(Object.keys(publishers)) as any
+        })
+    }
+    for await (const msg of messages) {
+        await publishers[msg.publisher].publish(msg, streamId)
+    }
+    await Promise.all(Object.values(publishers).map((publisher) => publisher.close()))
+    return messages
+}
+
+describe('multiple publisher plugins', () => {
+
+    let tracker: Tracker
+    let broker: Broker
+    let privateKey: string
+    let streamId: string
+
+    beforeAll(async () => {
+        privateKey = await fetchPrivateKeyWithGas()
+        tracker = await startTestTracker(trackerPort)
+        const client = await createClient(tracker, privateKey)
+        const stream = await createTestStream(client, module)
+        streamId = stream.id
+        stream.grantPermissions({
+            permissions: [StreamPermission.SUBSCRIBE],
+            public: true
+        })
+        await client.destroy()
+    })
+
+    afterAll(async () => {
+        await tracker.stop()
+    })
+
+    beforeEach(async () => {
+        broker = await startBroker({
+            privateKey,
+            trackerPort,
+            httpPort,
+            extraPlugins: {
+                mqtt: {
+                    port: mqttPort
+                },
+                websocket: {
+                    port: wsPort
+                },
+                http: {}
+            }
+        })
+    })
+
+    afterEach(async () => {
+        await broker.stop()
+    })
+
+    it('subscribe by StreamrClient', async () => {
+
+        const receivedMessages: Queue<object> = new Queue()
+        const subscriber = await createClient(tracker, fastPrivateKey())
+        await subscriber.subscribe(streamId, (message: object) => {
+            receivedMessages.push(message)
+        })
+        // TODO: better waiting
+        await wait(3000)
```

Comment:
Let's figure out a better waiting strategy.

We can either use `NetworkNodeStub#getNeighborsForStreamPart` or we could consult tracker on what nodes are in which stream partition.